Formula/hdf5.rb

```diff
@@ -56,10 +56,7 @@ def install
     system "./configure", *args
 
     # Avoid shims in settings file
-    inreplace "src/libhdf5.settings", %r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{ENV.cc}}, ENV.cc
-    if OS.linux?
-      inreplace "src/libhdf5.settings", %r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{Regexp.escape(ENV.cxx)}}, ENV.cxx
-    end
+    inreplace "src/libhdf5.settings", Superenv.shims_path/ENV.cc, ENV.cc
```

Comment:
Looks like the Linux case got dropped here.

```suggestion
    inreplace "src/libhdf5.settings", Superenv.shims_path/ENV.cc, ENV.cc
    inreplace "src/libhdf5.settings", Superenv.shims_path/ENV.cxx, ENV.cxx if OS.linux?
```